examples/docs_beta_snippets/docs_beta_snippets/guides/data-modeling/partitioning/time_based_partition_dependencies.py

```diff
@@ -0,0 +1,67 @@
+import datetime
+import os
+
+import pandas as pd
+
+import dagster as dg
+
+# Create the PartitionDefinition
+daily_partitions = dg.DailyPartitionsDefinition(start_date="2024-01-01")
+weekly_partitions = dg.WeeklyPartitionsDefinition(start_date="2024-01-01")
+
+
+# Define the partitioned asset
+@dg.asset(
+    partitions_def=daily_partitions,
+    # highlight-start
+    automation_condition=dg.AutomationCondition.on_cron(cron_schedule="0 1 * * *"),
+    # highlight-end
+)
+def daily_sales_data(context: dg.AssetExecutionContext):
+    date = context.partition_key
+    # Simulate fetching daily sales data
+    df = pd.DataFrame({"date": [date], "sales": [1000]})
+
+    os.makedirs("data/daily_sales", exist_ok=True)
+    filename = f"data/daily_sales/sales_{date}.csv"
+    df.to_csv(filename, index=False)
+
+    context.log.info(f"Daily sales data written to {filename}")
+
+
+@dg.asset(
+    partitions_def=weekly_partitions,
+    # highlight-start
+    automation_condition=dg.AutomationCondition.eager(),
+    # highlight-end
+    deps=[daily_sales_data],
+)
+def weekly_sales_summary(context: dg.AssetExecutionContext):
+    week = context.partition_key
+    # highlight-start
+    partition_key_range = context.asset_partition_key_range_for_input(
```

Comment:
is this the right api? "input" doesn't look good to me because this is the deps path (no i/o mention path)

Reply:
We switched hooli to use https://github.com/dagster-io/hooli-data-eng-pipelines/blob/master/hooli_data_eng/assets/dbt_assets.py#L122

Reply:
We use the same one in DOP. No input/output plz